Deprecate all JavaScript instance properties the except `init` method (#3499)
alphagov/govuk-frontend: CHANGELOG.md

### Deprecated features

#### Stop using JavaScript API properties other than the `init` method

> All JavaScript properties in the API for our JavaScript components except for the `init` method are now deprecated, and will be made private in v5.0.

Reviewer: ```suggestion
We have deprecated all JavaScript properties in the API for our JavaScript components, except for the `init` method. We will make all deprecated JavaScript properties private in v5.0.
```

Author: What about this, to avoid some of the repetition? (Especially 'JavaScript' which was already in there twice…)

> We have deprecated all of the JavaScript properties in the API except for the `init` method. We will make them private in v5.0.

Reviewer: I repeated as the last thing mentioned in the previous sentence in the init method, which means grammatically 'them' would refer to the init method. I shall shuffle things...

Reviewer: I've amended my suggestion - does that work?

Author: I think the version with the repetition was better – this one seems to slightly contradict itself. What about:

> We have deprecated all of the JavaScript properties in the API, except for the `init` method for each component. We will make all the deprecated JavaScript properties private in v5.0.

Author: Trying to break it down:

- We have deprecated all of the JavaScript properties in the API.
- The exception to this is the `init` method for each component, which remains public.

With the additional comma I think it would read as:

- We have deprecated all of the JavaScript properties in the API for every component.
- The exception to this is the `init` method, which remains public.

Which isn't quite right – every component has an `init` method, which will remain public; and there are also parts of the API that aren't part of any component.

Reviewer: So I've changed the suggestion to your 3 sentences.

> If you're calling a method other than `init`, [please tell us by filling in this form](https://docs.google.com/forms/d/e/1FAIpQLSfmH2AitMeouXqB0FWC5p5e6y1TSiFCjmJ8DrVuwfmpRGCaWw/viewform?usp=sf_link). We'll use this information when prioritising future additions to the public API.

Author: Please review the content and structure of the linked Google Form as well. I've pinged @claireashworth to get her thoughts on it too.
